bofire/data_models/features/categorical.py

```diff
@@ -401,9 +404,33 @@ def validate_experimental(self, values: pd.Series) -> pd.Series:
             )
         return values
 
+    def __str__(self) -> str:
+        return "CategoricalOutputFeature"
+
     def to_dict(self) -> Dict:
         """Returns the catergories and corresponding objective values as dictionary"""
-        return dict(zip(self.categories, self.objective))
+        return dict(zip(self.categories, self.objective.desirability))
```

Comment:
This would mean we set up functions within the objective which take in the categories and do some processing. I think it's still unclear to me why the categories cannot be directly passed to the objective upon instantiation